frontend/pages/hubs/[hubUrl]/index.tsx

```diff
@@ -0,0 +1,213 @@
+  climateHub: string;
+  find_suitable_climate_protection_commitment: string;
+  find_fellow_campaigners_for_climate_protection_idea: string;
+  coordinates_the_climateHub: string;
+  is_there_for_you: string;
+  [key: string]: string;
+}
+
+interface NotFoundPageProps {
+  texts: TextsType;
+  link: string;
+  showHeader?: boolean;
+}
+
+interface LandingPageProps {
+  hubAmbassador: HubAmbassador | null;
+  hubData: HubData | null;
+  hubUrl?: string;
+}
+
+const NotFoundPage: React.FC<NotFoundPageProps> = ({ texts, link, showHeader }) => {
+  return (
+    <>
+      {showHeader ? (
+        <WideLayout>
+          <PageNotFound
+            itemName="landing page"
+            returnText={texts.return_to_hubs}
+            returnLink={link}
+          />
+        </WideLayout>
+      ) : (
+        <PageNotFound itemName="landing page" returnText={texts.return_to_hubs} returnLink={link} />
+      )}
+    </>
+  );
+};
+
+//for description we need the Ambassador name
+export async function getServerSideProps(ctx: any) {
+  const hubUrl = ctx?.params?.hubUrl as string | undefined;
+
+  if (!hubUrl) {
+    return {
+      props: {
+        hubAmbassador: null,
+        hubData: null,
+        hubUrl: null,
+      },
+    };
+  }
+
+  const [hubAmbassador, hubData] = await Promise.all([
+    getHubAmbassadorData(hubUrl, ctx.locale),
+    getHubData(hubUrl, ctx.locale),
+  ]);
+
+  return {
+    props: {
+      hubAmbassador,
+      hubData,
+      hubUrl,
+    },
+  };
+}
+
+const getHubAmbassadorData = async (
+  hubUrl: string,
+  locale: LocaleType
+): Promise<HubAmbassador | null> => {
+  try {
+    const resp = await apiRequest({
+      method: "get",
+      url: `/api/hubs/${hubUrl}/ambassador/`,
+      locale,
+    });
+    return resp.data;
+  } catch (err: any) {
+    console.error(
+      "Error fetching hub ambassador data:",
+      err.response?.data?.detail || err.message || err
+    );
+    return null;
+  }
+};
+
+const getHubData = async (url_slug: string, locale: LocaleType): Promise<HubData | null> => {
+  try {
+    const resp = await apiRequest({
+      method: "get",
+      url: `/api/hubs/${url_slug}/`,
+      locale,
+    });
+    return resp.data;
+  } catch (err: any) {
+    console.error("Error fetching hub data:", err.response?.data?.detail || err.message || err);
+    return null;
+  }
+};
+
+const LandingPage: React.FC<LandingPageProps> = ({ hubAmbassador, hubData, hubUrl }) => {
+  const { locale } = useContext(UserContext);
+  const texts = getTexts({ page: "landing_page", locale: locale }) as TextsType;
+  const [DevlinkComponent, setDevlinkComponent] = useState<DevlinkComponentType>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    const loadComponent = async () => {
+      if (!hubData?.landing_page_component) {
+        setIsLoading(false);
+        return;
+      }
+
+      try {
+        // Identify the correct component name based on locale
```

Comment:
I think it would be good to explain in the comments how the names of the components are structured, maybe with an example. That would make it easier for new contributors to read the code